src/main/java/com/simprints/simface/quality/MlKitFaceDetectionProcessor.kt

```diff
@@ -136,4 +146,258 @@ internal class MlKitFaceDetectionProcessor() : FaceDetectionProcessor {
 
         return (leftEyeScore + rightEyeScore) / 2.0
     }
+
+    fun buildLandmarks(face: Face): FacialLandmarks? {
+        val leftEyeLandmark = face.getLandmark(FaceLandmark.LEFT_EYE)
+        val leftEyeContour = face.getContour(FaceContour.LEFT_EYE)
+        val leftEye = if (leftEyeLandmark != null) {
+            leftEyeLandmark.position
+        } else if (leftEyeContour != null && leftEyeContour.points.size > 4) {
+            leftEyeContour.points[4]
+        } else {
+            return null
+        }
+
+        val rightEyeLandmark = face.getLandmark(FaceLandmark.RIGHT_EYE)
+        val rightEyeContour = face.getContour(FaceContour.RIGHT_EYE)
+        val rightEye = if (rightEyeLandmark != null) {
+            rightEyeLandmark.position
+        } else if (rightEyeContour != null && rightEyeContour.points.size > 4) {
+            rightEyeContour.points[4]
+        } else {
+            return null
+        }
+
+        val noseLandmark = face.getLandmark(FaceLandmark.NOSE_BASE)
+        val noseContour = face.getContour(FaceContour.NOSE_BRIDGE)
+        val nose = if (noseLandmark != null) {
+            noseLandmark.position
+        } else if (noseContour != null && noseContour.points.isNotEmpty()) {
+            noseContour.points.last()
+        } else {
+            return null
+        }
+
+        val lowerLipContour = face.getContour(FaceContour.LOWER_LIP_BOTTOM)
+        val mouthLeftLandmark = face.getLandmark(FaceLandmark.MOUTH_LEFT)
+        val mouthLeft = if (mouthLeftLandmark != null) {
+            mouthLeftLandmark.position
+        } else if (lowerLipContour != null && lowerLipContour.points.isNotEmpty()) {
+            lowerLipContour.points.last()
+        } else {
+            return null
+        }
+
+        val mouthRightLandmark = face.getLandmark(FaceLandmark.MOUTH_RIGHT)
+        val mouthRight = if (mouthRightLandmark != null) {
+            mouthRightLandmark.position
+        } else if (lowerLipContour != null && lowerLipContour.points.isNotEmpty()) {
+            lowerLipContour.points.first()
+        } else {
+            return null
+        }
+
+        return FacialLandmarks(
+            Point2D(leftEye.x, leftEye.y),
+            Point2D(rightEye.x, rightEye.y),
+            Point2D(nose.x, nose.y),
+            Point2D(mouthLeft.x, mouthLeft.y),
+            Point2D(mouthRight.x, mouthRight.y)
+        )
+    }
+
+
+    override fun warpAlignFace(face: FacialLandmarks, inputImage: Bitmap): Bitmap {
+
+        val ref = arrayOf(
```

Comment:
Any set of magic numbers requires a full descriptive name and a comment that explains them.